lang/cs/Org.Apache.REEF.IO/PartitionedData/FileSystem/FileSystemInputPartition.cs

```diff
@@ -54,26 +56,73 @@ internal sealed class FileSystemInputPartition<T> : IInputPartition<T>, IDisposa
             _id = id;
             _fileSystem = fileSystem;
             _fileSerializer = fileSerializer;
-            _filePaths = filePaths;
             _tempFileCreator = tempFileCreator;
-            _isInitialized = false;
-            _copyToLocal = copyToLocal;
+            _filePaths = filePaths;
+
+            if (!copyToLocal)
+            {
+                // Implies that the files are already local.
+                _localFiles = Optional<ISet<string>>.Of(filePaths);
+            }
+            else
+            {
+                _localFiles = Optional<ISet<string>>.Empty();
+            }
         }
 
         public string Id
         {
             get { return _id; }
         }
 
-        private void Initialize()
+        public void Cache()
```

Comment:
No, that iteration is done as part of DeSerialize currently. Where to desrialize and get the type of T all depends on user and default implementations we provide. For example, we might have tw implementation of HDD cache where one involved simple downloading to disk while some other customized implementation might involve reading from remote row by row, converting in to some intermediate format and then writing to disk. Similarly, Memory cache might also have two implementations - in one we simply load the full raw data in memory while in some customized implementation we might also do parsing and then load only parsed data in memory. It all depends on the user now. And we will provide some commonly known and used interpretations of these cache.